### DISCUSSION TITLE
How to create rodent structural template using RABIES

### DISCUSSION THREAD

**Discussion by MRImagingLi** [CoBrALab/RABIES#270] [Q&A]
Just wondering how to create a structural template from your own T1 or T2 highres images from Rabies using external template or no external template?

Alex

**Comment by gdevenyi** [comment]
RABIES is intended primarily for preprocessing fMRI data. If you're interested specifically with building average structural templates, please take a look at https://github.com/CoBrALab/optimized_antsMultivariateTemplateConstruction which is the primary tool RABIES uses inside to achieve good coregistration.

**Reply by MRImagingLi** [reply]
Hi Gabriel, 

Thanks for the answer. Since [optimized_antsMultivariateTemplateConstruction] is the primary tool of RABIES used for coregistration, there should be a way to use it directly inside RABIES for structural template creation. I used RABIES image, and do not want to install another copy of [optimized_antsMultivariateTemplateConstruction] on my computer anymore. 

In the future, your program at least should be used for:
1. High resolution structural image registration
2. Bold images analysis
3. DTI images analysis

They can be analysized together or separately.

If you can have a way to use the TemplateConstrustion for anatomical image registration, that will be great. Thanks

Alex


1.

**Reply by Gab-D-G** [reply]
Hi Alex, if you ran RABIES with functional images, the generated anatomical template should be at least accessible through in the unbiased_template_datasink/ folder. RABIES cannot be currently run without BOLD images, but further updates could easily adopt the current workflow to handle anatomical-only preprocessing. However, you could currently run optimized_antsMultivariateTemplateConstruction from your RABIES image without further installation. For instance with singularity, by using `singularity exec rabies_image.sif template_construction.sh` you could execute a .sh script running the template construction using the RABIES image, without running the actual functional preprocessing pipeline. You could use the RABIES command for optimized_antsMultivariateTemplateConstruction https://github.com/CoBrALab/RABIES/blob/6f19f204c984ed135c101936e60422ed2e06651a/rabies/preprocess_pkg/commonspace_reg.py#L583 as reference to run the customize your own command, and running the same template construction. You would need to handle inhomogeneity correction beforehand however.

**Reply by MRImagingLi** [reply]
Thanks Gab, I will run a test sometime. 

Alex
